src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -178,12 +204,13 @@ private void Reset(bool isDisposing = false)
 
             if (!isDisposing)
             {
+                _solutionLoadedTcs = new TaskCompletionSource<bool>();
```

Comment:
Consider it imperative that you complete all Tasks that you start. I don't see an obvious guarantee that `_solutionLoadedTcs.Task` has completed before you blow away your ability to complete it with this line. So before this line, I suggest you add a line like this `_solutionLoadedTcs.TrySetCanceled()` or similar, so that if anyone is awaiting the Task, they won't hang forever.